src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -16,6 +16,22 @@ public PackageSourceMappingProvider(ISettings settings)
             _settings = settings ?? throw new ArgumentNullException(nameof(settings));
         }
 
+        /// <summary>
+        /// Supports the disabling of saving to disk for any <see cref="ISettings"/> changes.
+        /// </summary>
+        /// <param name="shouldSkipSave">True to avoid saving any changes to disk and only modify the <see cref="ISettings"/> in memory.
+        /// Default is false.</param>
+        public PackageSourceMappingProvider(ISettings settings, bool shouldSkipSave)
+            : this(settings)
+        {
+            ShouldSkipSave = shouldSkipSave;
+        }
+
+        /// <summary>
+        /// Avoid saving to disk and only modify the <see cref="ISettings"/> in memory.
+        /// </summary>
+        public bool ShouldSkipSave { get; private set; }
```

Comment:
This can be read-only?
```suggestion
        public bool ShouldSkipSave { get; }
```